### DISCUSSION TITLE
Custom UI - ComboBox with ListItems that publish to a Windows Registry key.

### DISCUSSION THREAD

**Discussion by JClayton-TAIT** [wixtoolset/issues#7449] [Questions]
I am trying to figure out (in either v3 or v4) how to create a custom dialog that has a simple ComboBox with 4 ListItem's. The value of this ComboBox would get published to a Windows Registry key during the installation.

**Comment by robmen** [comment]
If using MSI UI, I *think* this is a matter of adding temporary rows to the correct table. You probably need to do so before the page is visible as well because I don't think custom actions can modify the content of the current page.

If you're using a custom Bootstrapper Application then use your UI framework of choice.

**Reply by JClayton-TAIT** [reply]
How would you go about passing the chosen value of a ComboBox to a CustomAction DLL then?

**Reply by BMurri** [reply]
> How would you go about passing the chosen value of a ComboBox to a CustomAction DLL then?

Via an MSI property. If you are referring to a ComboBox in a Windows Installer UI the name of the property your custom action should look at will likely be the value of the `Property` attribute of the `ComboBox` element.

**Reply by JClayton-TAIT** [reply]
I have the following as an example, but not sure what the C# DLL should look like exactly. I don't have the project examples available in Visual Studio to add a Custom Action for some reason.
Is the below even close?
`
<?xml version="1.0" encoding="UTF-8"?>
<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
  <Product Id="*" Name="MyProduct" Language="1033" Version="1.0.0.0" Manufacturer="MyCompany" UpgradeCode="PUT-GUID-HERE">
    <Package InstallerVersion="200" Compressed="yes" />

    <UI>
      <Dialog Id="ComboBoxDialog" Width="370" Height="270" Title="ComboBox Dialog">
        <Control Id="ComboBox" Type="ComboBox" X="40" Y="90" Width="290" Height="18" Property="COMBOBOXPROP">
          <ComboBox Property="COMBOBOXPROP">
            <ListItem Text="Option 1" Value="1" />
            <ListItem Text="Option 2" Value="2" />
            <ListItem Text="Option 3" Value="3" />
            <ListItem Text="Option 4" Value="4" />
          </ComboBox>
          <Publish Property="COMBOBOXPROP" Value="[ComboBox]" Order="1">1</Publish>
          <Publish Event="EndDialog" Value="Return" Order="2">1</Publish>
        </Control>
      </Dialog>
    </UI>

    <Binary Id="MyCustomAction" SourceFile="MyCustomAction.dll" />
    <CustomAction Id="ExecuteCustomAction" BinaryKey="MyCustomAction" DllEntry="MyCustomAction" Execute="differed" />

    <InstallExecuteSequence>
      <Custom Action="ExecuteCustomAction" Before="InstallFinalize">COMBOBOXPROP</Custom>
    </InstallExecuteSequence>
  </Product>
</Wix>
`

**Reply by BMurri** [reply]
If the custom action is written in C#, you will probably need to have your SourceFile point to "MyCustomAction.CA.dll". Inside of your custom action, you will get the value by accessing `session["COMBOBOXPROP"]` which will return a string.
